### ISSUE TITLE
Allow Blocks to be used in Jobdescription

### ISSUE THREAD

**Issue by masteradhoc** [Automattic/WP-Job-Manager#2655]
**Describe the bug**
Currently WPJM Users that are using a FSE or Hypbrid Theme with Gutenberg are restricted to use the classic block. See
![image](https://github.com/Automattic/WP-Job-Manager/assets/6242098/73fcac21-5328-48a6-b716-dc36a5c949be)

Even if you delock the block a convert to blocks can be clicked but nothing happens
![image](https://github.com/Automattic/WP-Job-Manager/assets/6242098/eae28d46-1a49-4b83-b682-1746c3bfe3da)

**To Reproduce**
1. Install GeneratePress Theme & WPJM
2. Create a Job
3. Start typing Text

**Expected behavior**
Inside the post description i should be able to use all blocks as on a post/page

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WP Job Manager plugin is active.
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
- WordPress Version: 6.4.1
- WP Job Manager Version: 2.1.1
- PHP Version: 8.2
- Other important details: -

**Comment by yscik** [comment]
We plan to work on this in the foreseeable future. The main thing here is that if we enable blocks in the admin, they also need to work in the frontend editor. Fortunately it's possible now to also use the Gutenberg editor there.

**Comment by ouun** [comment]
@yscik I am wondering if that behavior can be bypassed in the meantime to use the Block Editor.
We are not planning to use the front-end form and therefore do not need to support the blocks there.

**Comment by yscik** [comment]
Yes, it's possible. Job Manager forces the template for job listings to be a single "Classic block". This snippet can remove that configuration:

```php
<?php

add_filter( 'register_post_type_job_listing', 'wpjm_snippet__job_listing_enable_blocks' );

function wpjm_snippet__job_listing_enable_blocks( $args ) {
   	$args['template'] = [];
   	$args['template_lock'] = false;

    return $args;
}
```
